src/hotspot/share/utilities/elfFile.hpp

```diff
@@ -861,6 +861,8 @@ class DwarfFile : public ElfFile {
     bool does_offset_match_entry(uintptr_t previous_address, uint32_t previous_file, uint32_t previous_line);
     void print_and_store_prev_entry(uint32_t previous_file, uint32_t previous_line);
     bool get_filename_from_header(uint32_t file_index, char* filename, size_t filename_len);
+    static void strip_path_prefix(char* filename, const size_t filename_len);
+    static void copy_dwarf_filename_to_filename(char* src, size_t src_len, char* dst, size_t dst_len);
```

Comment:
Stupid question, do these have to be exposed? Or could they be just static helpers in elfFile.cpp?

Reply:
In theory, they don't need to be exposed in the sense of being declared in the header. But in terms of readability, I've decided to put them in the private block of class `LineNumberProgram` which is the only user of these methods. What would be the advantages of moving them completely to `elfFile.cpp` as static helper functions?

Reply:
Oh, I generally just prefer to keep things locally if possible. Slim interfaces, less polluted global namespace, possibly (though not here) less include deps. Don't worry, if you prefer it this way, keep it in.

Reply:
I understand. I don't have a strong preference here but to be consistent with the existing DWARF parser code, I'd rather keep it like that.